packages/ngtools/webpack/src/ngcc_processor.ts

```diff
@@ -189,7 +193,10 @@ export class NgccProcessor {
 
     const timeLabel = `NgccProcessor.processModule.ngcc.process+${moduleName}`;
     time(timeLabel);
-    this.compilerNgcc.process({
+
+    // Use a void expression to suppress @typescript-eslint/no-floating-promises as this calls
+    // ngcc synchronously.
+    void this.compilerNgcc.process({
```

Comment:
This method doesn’t return a promise.

Reply:
ESLint gets confused because of the conditional return type (based on the `async: true` option). I'm not even sure why this is an issue now, does ESLint only report errors for touched files?

Reply:
No, it reports errors on all files, which is even more strange.

Reply:
Hmmm, I just ran `yarn lint` locally after reverting this change and there was no error then. So now it's even more weird why [the initial CI run](https://app.circleci.com/pipelines/github/angular/angular-cli/23830/workflows/2a1cc71b-a214-46b9-b6f2-bdeaec517b4b/jobs/314149) errored out on this. I can drop the commit if you'd prefer.

Reply:
Actually, I do get it now. There was a compile error in that commit where the options object included a `null` type for `basePath`, which is not accepted. So it didn't type check and the return type couldn't actually be narrowed to be non-Promise. So I'm dropping this change :-)